draco/synthesis/expand_perturbed.py

```diff
@@ -0,0 +1,598 @@
+        perturbation_list = containers.BeamPerturbation(freq = freqmap, input = pertlistlen,
+                            distributed = True, prod = desired_prod, comm = map_.comm)
+        # Add beam perturbation values to container.
+        perturbation_list.pert[:] = perturbations
+
+        return perturbation_list
+
+class ExpandPerturbedProducts(task.SingleTask):
+    """Un-wrap collated products to full triangle.
+    Combine unperturbed and perturbed beam components into one set of output products.
+    """
+
+    def setup(self, telescope):
+        """Get a reference to the telescope class.
+
+        Parameters
+        ----------
+        tel : :class:`drift.core.TransitTelescope`
+            Telescope object.
+        """
+        self.telescope = telescope
+
+    def process(self, sstream, map_,perturbations):
+    #def process(self,sstream,map_,perturbations=0)
+        """Transform a sidereal stream to having a full product matrix.
+
+        Parameters
+        ----------
+        sstream : :class:`containers.SiderealStream`
+            Sidereal stream to unwrap.
+
+        Returns
+        -------
+        new_sstream : :class:`containers.SiderealStream`
+            Unwrapped sidereal stream with unperturbed and perturbed components of products combined.
+        """
+        tel = self.telescope
+
+        # Load and redistribute perturbations from input list.
+        # Generally, this list will come from GeneratePertubation, but it may
+        # also be any other BeamPerturbation container generated any other way.
+        perturbations.redistribute('freq')
+        pert_arr = perturbations.pert[:]
+        # Select only set 0 - this avoids the whole MPI Communicator thing.
+        feed_perts = pert_arr[0,:]
+
+        # Redistribute sstream over freq
+        sstream.redistribute('freq')
+
+        # Define total ninput based on sstream
+        ninput = len(sstream.input)
+
+        # Determine ninput per perturbation based on ninput/n perturbations.
+        ninputperpert = len(sstream.input) / tel.npert
+
+        # Define array with the size of all of the products from sstream.
+        #all_prod = np.array([ (fi, fj) for fi in range(ninput) for fj in range(fi, ninput)])
+
+        # Define array with the size of the desired number of total products from sstream.
+        desired_prod = np.array([ (fi, fj) for fi in range(ninputperpert) for fj in range(fi, ninputperpert)])
+
+        # Set l max, m max, number of frequencies,  number of polarizations, and sky resolution in the same manner as sstream.
+        # We aren't changing  any of them, but we need have them defined here to make new_stream.
+        lmax = tel.lmax
+        mmax = tel.mmax
+        nfreq = tel.nfreq
+        npol = tel.num_pol_sky
+
+        # Set the minimum resolution required for the sky.
+        ntime = 2 * mmax + 1
+
+        freqmap = map_.index_map['freq'][:]
+        row_map = map_.map[:]
```

Comment:
I think you never use this variable 'row_map'. 
This seems to be true in all tasks that define it.